src/acl/Checklist.cc

```diff
@@ -325,8 +325,8 @@ ACLChecklist::fastCheck()
 void
 ACLChecklist::calcImplicitAnswer()
 {
-    const auto lastAction = accessList ?
-                            accessList->lastAction() : Acl::Answer(ACCESS_DUNNO);
+    const auto& lastAction = accessList ?
+                             accessList->lastAction() : Acl::Answer(ACCESS_DUNNO);
```

Comment:
The implementation of `lastAction` has two outputs. Only one of which meets the copy-ellision criteria.
```c++
    if (actions.empty())
        return ACCESS_DUNNO; // enumerator => copy-ellision
    return actions.back(); // non-const rvalue => copy is mandatory
```

Reply:
> The implementation of lastAction has ...

The _implementation_ of lastAction() is irrelevant for the copy elision requirements.

Reply:
Is there any way to validate this? E.g. instrumenting the copy constructor and see if it gets invoked?
FWIW, I've tried running a more recent version of Coverity Scan, and identified more of these defects, not fewer.
Could it be that the compiler has a different understanding or nonconformant implementation?

Reply:
> Is there any way to validate this?

Already done at https://github.com/squid-cache/squid/pull/1955#discussion_r1863700178 AFAICT. If that comment is not the validation you are after, then please define "this".

> E.g. instrumenting the copy constructor and see if it gets invoked? 

Similar checks were performed in tests linked from the same https://github.com/squid-cache/squid/pull/1955#discussion_r1863700178. If those are not the tests you want, then please adjust them.

> Could it be that [Coverity?] has a different understanding or nonconformant implementation?

Yes, it could be. That is essentially one of the two theories proposed at the same https://github.com/squid-cache/squid/pull/1955#discussion_r1863700178. FWIW, I would expect to see more and more such false defects as we improve Squid code until Coverity [default] implementation catches up (or we learn how to configure Coverity better to force the right behavior).

Reply:
I've done a different [test](https://godbolt.org/z/4sxzrK7aa) and it shows a different result from yours. Could you double check?

Reply:
> I've done a different [test](https://godbolt.org/z/4sxzrK7aa) and it shows a different result from yours. Could you double check?

AFAICT, your test does not show a different (or same) result compared to mine. It shows an unrelated result -- the two test results cannot be compared on inequality (or equality) as they are testing different things.

I do not see how your test of what looks like lastAction() _implementation_ is relevant to this change request about lastAction() _calling_ code. Could you please clarify why test code Y when Coverity defect, this PR, and this change request are all about code X (that does not even _see_ code Y)?